cassiopeia/solver/dissimilarity_functions.py

```diff
@@ -161,14 +161,22 @@ def hamming_similarity_normalized_over_missing(
 
 
 @numba.jit(nopython=True)
-def hamming_distance(s1: np.array(int), s2: np.array(int)) -> int:
+def hamming_distance(
+    s1: np.array(int),
+    s2: np.array(int),
+    ignore_missing_state: bool = False,
+    missing_state_indicator: int = -1,
+) -> int:
     """Computes the vanilla hamming distance between two samples.
 
     Counts the number of positions that two samples disagree at.
```

Comment:
Should we add to the name and docstring that this ignores missing data?

Reply:
But it doesn't ignore missing data all the time, so I don't want to add it to the function name.

I've added some more details to the docstring.